ibis/backends/tests/test_client.py

```diff
@@ -1629,3 +1629,8 @@ def test_from_connection(con, top_level):
     new_con = backend.from_connection(getattr(con, CON_ATTR.get(con.name, "con")))
     result = int(new_con.execute(ibis.literal(1, type="int")))
     assert result == 1
+
+
+def test_table_not_found(con):
+    with pytest.raises(com.TableNotFound, match="Table not found"):
+        con.table("foo")
```

Comment:
```suggestion
        con.table(util.gen_name("table_not_found"))
```

Let's use a name that no human would reasonably author by accident. You might need `from ibis import util` for this as well.